apps/arweave/src/ar_mining_server.erl

```diff
@@ -293,6 +264,16 @@ refresh_workers(State) ->
 	WorkersBySession2 = refresh_workers(SessionsToAdd, SessionsToRemove2, WorkersBySession),
 	State#state{ workers = WorkersBySession2 }.
 
+get_current_session_keys() ->
+	case ar_pool:is_client() of
+		false ->
+			{CurrentSessionKey, CurrentSession} = ar_nonce_limiter:get_current_session(),
```

Comment:
What do you think about moving the session logic into `ar_nonce_limiter`? 

Like instead of doing the `case ar_pool:is_client()` when calling out to `ar_nonce_limiter`, have ar_nonce_limiter do the `case ar_pool:is_clien()` itself and return the correct sessions. This could localize the complexity of determining the correct sessions to the `ar_nonce_limiter` module and allow callers of the `ar_nonce_limiter` to be agnostic to how the sessions are determined